rumqttc/src/mqttbytes/mod.rs

```diff
@@ -13,33 +13,50 @@ pub mod v4;
 pub use topic::*;
 
 /// Error during serialization and deserialization
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, thiserror::Error)]
 pub enum Error {
+    #[error("Expected Connect, received: {0:?}")]
     NotConnect(PacketType),
+    #[error("Unexpected Connect")]
     UnexpectedConnect,
+    #[error("Invalid Connect return code: {0}")]
     InvalidConnectReturnCode(u8),
-    InvalidReason(u8),
```

Comment:
looking at the final diff…

Not sure what happened here but are some cases missing now? It still compiles so they weren’t used?

Reply:
I removed them as they were deemed unnecessary